source/utilities.tex

```diff
@@ -15967,6 +15972,396 @@
+template<class T, class U, class... Args>
+  explicit move_only_function(in_place_type_t<T>, initializer_list<U> ilist, Args&&... args);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{VT} be \tcode{decay_t<T>}.
+
+\pnum
+\constraints
+\begin{itemize}
+\item
+\tcode{is_constructible_v<VT, initializer_list<U>\&, ArgTypes...>} is
+\tcode{true}, and
+\item
+\tcode{\exposid{is-callable-from}<VT>} is \tcode{true}.
+\end{itemize}
+
+\pnum
+\mandates
+\tcode{VT} is the same type as \tcode{T}.
+
+\pnum
+\expects
+\tcode{VT} meets the \oldconcept{Destructible} requirements, and
+if \tcode{is_move_constructible_v<VT>} is \tcode{true},
+\tcode{VT} meets the \oldconcept{MoveConstructible} requirements.
+
+\pnum
+\ensures
+\tcode{*this} has a target object of type \tcode{VT}
+direct-non-list-initialized with
+\tcode{ilist, std::for\-ward<ArgTypes>(args)...}.
+
+\pnum
+\throws
+Any exception thrown by the initialization of the target object.
+May throw \tcode{bad_alloc} unless \tcode{VT} is
+a function pointer or a specialization of \tcode{reference_wrapper}.
+\end{itemdescr}
+
+\indexlibrarymember{operator=}{move_only_function}%
+\begin{itemdecl}
+move_only_function& operator=(move_only_function&& f);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to: \tcode{move_only_function(std::move(f)).swap(*this);}
+
+\pnum
+\returns
+\tcode{*this}.
+\end{itemdescr}
+
+\indexlibrarymember{operator=}{move_only_function}%
+\begin{itemdecl}
+move_only_function& operator=(nullptr_t) noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Destroys the target object of \tcode{*this}, if any.
+
+\returns
+\tcode{*this}.
+\end{itemdescr}
+
+\indexlibrarymember{operator=}{move_only_function}%
+\begin{itemdecl}
+template<class F> move_only_function& operator=(F&& f);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to: \tcode{move_only_function(std::forward<F>(f)).swap(*this);}
+
+\pnum
+\returns
+\tcode{*this}.
+\end{itemdescr}
+
+\indexlibrarydtor{move_only_function}%
+\begin{itemdecl}
+~move_only_function();
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Destroys the target object of \tcode{*this}, if any.
+\end{itemdescr}
+
+\rSec4[func.wrap.move.inv]{Invocation}
+
+\indexlibrarymember{operator bool}{move_only_function}%
+\begin{itemdecl}
+explicit operator bool() const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\tcode{true} if \tcode{*this} has a target object, otherwise \tcode{false}.
+\end{itemdescr}
+
+\indexlibrarymember{operator()}{move_only_function}%
+\begin{itemdecl}
+R operator()(ArgTypes... args) @\cv{}@ @\placeholder{ref}@ noexcept(@\placeholder{noex}@);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\expects
+\tcode{*this} has a target object.
+
+\pnum
+\effects
+Equivalent to:
+\begin{codeblock}
+return @\placeholder{INVOKE}@<R>(static_cast<F @\placeholder{inv-quals}@>(f), std::forward<ArgTypes>(args)...);
+\end{codeblock}
+where \tcode{f} is the target object of \tcode{*this} and
```

Comment:
"an lvalue designating the target object of `*this`", perhaps? We do want to define its value category.